docs/DeveloperGuide.md

```diff
@@ -1,38 +1,287 @@
+| printByeMessage       | Prints the good bye message                                          |
+| printCreateStorage    | Prints the path to the storage that is created                       |
+| getSetCategory        | Capture user input when using the category function in certain modes |
+| printCat              | Prints the categories available in certain modes                     |
+| printCategoryNotFound | Prints when category is not found                                    |
+| print                 | Custom print function that accepts multiple strings                  |
+
+<u>Design Considerations</u>  
+This component was created as functionalities such as printing will be used in most of the classes in the PlanPal Application.  
+Consolidating all the print methods that are repeated reduces repetition in code and makes it easier to update when necessary.
+
+## Parser
+<u>Overview</u>
+This component handles the logic behind the application. The parser component consists of the parent `Parser` Class and 3 children that inherits the `Parser` Class. These 3 children are used when the respective modes are in play.  
+
+<u>Class Diagram for Parser Component</u>  
+![ParserClassDiagram.drawio.png](Images%28DG%29%2FParserClassDiagram.drawio.png)
+
+<u>Design Considerations</u>  
+The Parser class follows this structure as there are common commands between the different modes. Using inheritance prevents the repetition of code. Additionally, different parsers were created since different modes require different functionalities.
+
+## Mode: Contact Manager
+The class diagram below represents the contact book system
+
+<u>Class Diagram for Contact Manager</u>  
+![ContactClassDiagram.jpg](Images(DG)/ContactClassDiagram.jpg)
+
+
+The `Contact` class in PlanPal represents a user's contact, encapsulating details like name, phone, and email. It implements the Editable and Storeable interfaces, enabling contacts to be modified and stored.
+
+The `ContactManager` class in PlanPal manages a list of `Contact` objects. This class provides CRUD (Create, Read, Update, Delete) operations for handling contacts, along with advanced features for categorizing contacts. `ContactManager` implements the ListFunctions interface, allowing it to handle list-based operations in a structured way.
+
+The `ListFunctions` interface in the PlanPal application provides generic CRUD (Create, Read, Update, Delete) operations for list management. It standardizes common list manipulations, enabling reusable, modular operations across various object types and components.
+
+The `Editable`interface provides a standardized way to modify objects within PlanPal by defining an editing method, processEditFunction, which takes new values and updates object attributes accordingly.
+
+The `Storeable` interface standardizes essential storage-related functionalities within PlanPal, ensuring that objects implementing it can be managed and stored in a consistent manner.
+
+<u>Methods</u>
```

Comment:
These methods are very self-explanatory, you can consider omitting these descriptions.